src/App.js

```diff
@@ -1,22 +1,229 @@
+                                  {...dragProvided.dragHandleProps}
+                                  {...getItemProps({
+                                    onClick: () => this.removeItem(item),
+                                    item
+                                  })}
+                                >
+                                  {item.slug}
+                                </div>
+                                {dragProvided.placeholder}
+                              </div>
+                            )}
+                          </Draggable>
+                        ))
+                      }
+                    </div>
+                    {dropProvided.placeholder}
+                  </div>
+                )}
+                </Droppable>
+                )}/>
+
+          </DragDropContext>
+          <button onClick={() => this.setState({isOpen: !this.state.isOpen})}>open</button>
+          <hr/>
+
+          <Downshift
+            itemToString={itemToString}
+            isOpen={this.state.isOpen}
+            selectedItem={this.state.selectedLessons}
+            onStateChange={this.handleStateChange}
+          >
+            {({
+                getInputProps,
+                getItemProps,
+                isOpen,
+              }) => (
+              <div>
+                {!isOpen
+                  ? null
+                  : <div>
+                    <input {...getInputProps()} />
+                    {
+                      this.state.allLessons.map((item) => (
+                        <div key={item.slug} {...getItemProps({
+                          onClick: () => this.selectItem(item),
+                          item,
+                          style: {fontWeight: some(this.state.selectedLessons, item) ? 'bold' : 'normal'}
+                        })}>
+                          <input readOnly={true} type="checkbox" checked={some(this.state.selectedLessons, item)}/>{item.slug}
+                        </div>
+                      ))
+
+                    }
+                  </div>}
+              </div>
+            )}
+          </Downshift>
         </div>
       </StaticRouter>
     )
   }
 }
 
-export default App
\ No newline at end of file
+export default App
+
+function noop() {
+}
+
+const reorder = (list, startIndex, endIndex) => {
+  const result = Array.from(list);
+  const [removed] = result.splice(startIndex, 1);
+  result.splice(endIndex, 0, removed);
+
+  return result;
+}
+
+function unwrapArray(arg, defaultValue) {
+  arg = Array.isArray(arg) ? /* istanbul ignore next (preact) */ arg[0] : arg
+  if (!arg && defaultValue) {
+    return defaultValue
+  } else {
+    return arg
+  }
+}
+
+function composeEventHandlers(...fns) {
+  return (event, ...args) =>
+    fns.some(fn => {
+      fn && fn(event, ...args)
+      return event.defaultPrevented
+    })
+}
+
+function itemToString(item) {
+  return item.slug
+}
+
+class List extends React.Component {
```

Comment:
now that we have a working component above with no style applied at all, but mostly feature complete, the next step will be to clean it all up and create a "ContentSelector" component that fully encapsulates the behavior here and can be used and styled easily to suit different content selection needs across the app.